src/components/organisms/AdminVenueView/components/RoomCard/RoomCard.tsx

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import { Card } from "react-bootstrap";
+import { useAsyncFn } from "react-use";
+import Bugsnag from "@bugsnag/js";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import {
+  faUserFriends,
+  faEye,
+  faHandPointer,
+  faEyeSlash,
+  faBan,
+} from "@fortawesome/free-solid-svg-icons";
+
+import { RoomInput, upsertRoom } from "api/admin";
+
+import { Room, RoomType } from "types/rooms";
+import { VenueEvent } from "types/venues";
+
+import { WithId, WithVenueId } from "utils/id";
+import { openRoomUrl } from "utils/url";
+
+import { useUser } from "hooks/useUser";
+import { useRoom } from "hooks/useRoom";
+
+import { ButtonNG } from "components/atoms/ButtonNG/ButtonNG";
+import { EventCard } from "components/organisms/AdminVenueView/components/EventCard/EventCard";
+import { RoomIcon } from "components/organisms/AdminVenueView/components/RoomIcon/RoomIcon";
+import { PrettyLink } from "components/organisms/AdminVenueView/components/PrettyLink";
+
+import "./RoomCard.scss";
+
+interface RoomCardProps {
+  room: Room;
+  index: number;
+  venueId: string;
+  venueName: string;
+  events?: WithVenueId<WithId<VenueEvent>>[];
+}
+
+export const RoomCard: React.FC<RoomCardProps> = ({
+  room,
+  index,
+  venueId,
+  venueName,
+  events,
+}) => {
+  const { user } = useUser();
+
+  const { recentRoomUsers } = useRoom({ room, venueName });
+
+  const isRoomUnclickable = room.type === RoomType.unclickable;
+
+  const [{ loading: isTogglingRoom }, toggleRoom] = useAsyncFn(async () => {
+    if (!user) return;
+
+    try {
+      const roomValues: RoomInput = {
+        ...room,
+        isEnabled: !room.isEnabled,
+      };
+
+      await upsertRoom(roomValues, venueId, user, index);
+    } catch (e) {
+      Bugsnag.notify(e, (event) => {
+        event.addMetadata("AdminVenueView/RoomCard::updateRoom", {
+          venueId: venueId,
+          roomIndex: index,
+          newState: !room.isEnabled,
+        });
+      });
+    }
+  }, [index, room, user, venueId]);
+
+  const [
+    { loading: isTogglingClickability },
+    toggleRoomClickablility,
+  ] = useAsyncFn(async () => {
+    if (!user) return;
+
+    const roomType = isRoomUnclickable ? undefined : RoomType.unclickable;
```

Comment:
can we use `null` here instead of `undefined`? just because null is more explicit as a value

Reply:
What's the definition of explicit that `null` satisfies but `undefined` doesn't?

Do note the pattern of providing `undefined` as a value in `? :` is widely used throughout the project:
![sparkle-undefined-01](https://user-images.githubusercontent.com/79229621/128698980-a1abc10e-a36c-4648-865a-652b251c2af9.png)

Reply:
`undefined` is currently used across the project and is the default fallback value for JavaScript. Some of the firebase functions don't work with `null` but are fine with `undefined`.

Reply:
can we rename a variable to `updatedRoomType` or `newRoomType`, because currently it looks like it should be vise versa: if `isRoomUnclickable` then `RoomType.unclickable` else `undefined`?